### PR TITLE
Various dashboard fixes

### PR DESCRIPTION
**Pull request by arsenijesavic**
## Overview

**What features/fixes does this PR include?**

**Please provide the GitHub issue number**

Closes https://github.com/MetaFam/TheGame/issues/1209
Closes https://github.com/MetaFam/TheGame/issues/1210
Closes https://github.com/MetaFam/TheGame/issues/1223
Closes https://github.com/MetaFam/TheGame/issues/1224

## Follow up Improvement Ideas

- [ ] please list any improvement/ideas

## Implementation

**Describe technical (nontrivial / non-obvious) parts of your code**

**Side effects**

## Assets

[Include screenshots/videos if it makes reviewing easier.]

### CODE REVIEW

File: packages/web/utils/formatHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    X: 10,
+    IX: 9,
+    V: 5,
+    IV: 4,
+    I: 1,
+  };
+  let str = '';
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const i of Object.keys(roman)) {
```
Comment (alalonde) [inline packages/web/utils/formatHelpers.ts:26]: `Object.keys(roman).forEach` or `.reduce` ?   Would not require the eslint comment

File: packages/web/components/Player/Section/PlayerHero.tsx [hunk 2]
```diff
@@ -52,7 +55,7 @@ export const PlayerHero: React.FC<HeroProps> = ({ player, editing }) => {
       withoutBG
       p={[4, 2]}
     >
-      {isOwnProfile && !editing && (
+      {isOwnProfile && isMainNet && !editing && (
```
Comment (dysbulic) [inline packages/web/components/Player/Section/PlayerHero.tsx:58]: It seems like you're hiding the edit button, but not letting the player know they have to connect.

I need to look at the test instance to better understand this flow…

**Review by dysbulic — CHANGES_REQUESTED** [review]
Very good work in general. I had a couple comments on the season boundary generation and use of Unicode glyphs for the Roman numerals.

**Comment by dysbulic** [comment]
Yeah, I tried the test instance & it works like I thought it did. If I'm on the wrong chain I just lose the ability to edit without knowing that there's anything I should be doing. The `ConnectToProgress` button that was removed would have MetaMask prompt the user to automatically switch the network (and add it if necessary).

As well as being opaque, these changes also lose that functionality.

File: packages/web/components/EditProfileModal.tsx [hunk 3]
```diff
@@ -454,10 +452,6 @@ export const EditProfileModal: React.FC<ProfileEditorProps> = ({
     }
   };
 
-  if (chainId !== '0x1') {
-    return <ConnectToProgress />;
-  }
```
[the remark this answers is not in the document]
Comment (arsenijesavic) [reply packages/web/components/EditProfileModal.tsx:459]: @dysbulic It was due to [this issue ](https://github.com/MetaFam/TheGame/issues/1223). 
To leave it as is or to return ConnectToProgress ?

File: packages/web/components/Player/Section/PlayerHero.tsx (hunk 2 above)
Comment (arsenijesavic) [reply packages/web/components/Player/Section/PlayerHero.tsx:58]: @dysbulic It was due to [this issue ](https://github.com/MetaFam/TheGame/issues/1223) ?

File: packages/web/utils/formatHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    CD: 400,
+    C: 100,
+    XC: 90,
+    L: 50,
+    XL: 40,
+    X: 10,
+    IX: 9,
+    V: 5,
+    IV: 4,
+    I: 1,
```
[the remark this answers is not in the document]
Comment (arsenijesavic) [reply packages/web/utils/formatHelpers.ts:21]: @dysbulic I can't preview 100_000 and 50_000, its just square ?

File: packages/utils/src/xpHelpers.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      activeDate.getFullYear() % 4 === 0 ? 22 : 23,
+    ),
+
+    // WINTER_START
+    new Date(
+      activeDate.getFullYear(),
+      11,
+      activeDate.getFullYear() % 4 === 0 ? 20 : 21,
+    ),
+  ];
```
[the remark this answers is not in the document]
Comment (arsenijesavic) [reply packages/utils/src/xpHelpers.ts:61]: I agree about casting leapYear to the variable, but I feel that SEASON_START in this way is much harder to understand on the first glimpse ?

[the remark this answers is not in the document]
Comment (alalonde) [reply packages/utils/src/xpHelpers.ts:61]: I agree @arsenijesavic, I think your comments + breakout of calculations per spring is much more readable / understandable.

[the remark this answers is not in the document]
Comment (dysbulic) [reply packages/utils/src/xpHelpers.ts:61]: How about?

```javascript
const boundaries = [
  { month: 3, day: 19 },
  { month: 6, day: 20 },
  { month: 9, day: 22 },
  { month: 12, day: 20 },
]
const SEASON_START = (
  boundaries.map(({ month, day }) => (
    new Date(activeDate.getFullYear(), month - 1, day + (leapYear ? 0 : 1))
  )
)
```

This way you:
 • have the elements identified
 • can read everything on a single page
 • demonstrate that the month argument is zero-based
 • show the leap year calculation

Or you can leave the loop unrolled. I'm not too attached to it. I just tend toward loops when possible. At the very least, I'd switch the repeated `activeDate.getFullYear() % 4 === 0` for `leapYear` so the logic is clearer.

File: packages/web/components/EditProfileModal.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (dysbulic) [reply packages/web/components/EditProfileModal.tsx:459]: My ideal would be to have the edit button appear on the profile page if the user is logged in. When they click it, if they're on the wrong chain, prompt them to change.

Simply hiding the edit button if they're on the wrong chain will leave some users confused or simply not knowing they can edit.

File: packages/web/utils/formatHelpers.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (dysbulic) [reply packages/web/utils/formatHelpers.ts:21]: You just don't have those glyphs in your font, but it doesn't matter. If you copy and paste them it'll copy the codepoint even if you don't have the glyph.

I only put them in for completeness sake in any case. Season ↇ is thousands of years away. 😹

File: packages/web/components/Player/Section/PlayerHero.tsx (hunk 2 above)
Comment (dysbulic) [reply packages/web/components/Player/Section/PlayerHero.tsx:58]: As mentioned above, the best behavior would be to show the edit button, but prompt the user to change their chain if they're not on mainnet when they click it.

**Comment by alalonde** [comment]
@arsenijesavic Do you need any help with these requested changes?

**Comment by arsenijesavic** [comment]
> @arsenijesavic Do you need any help with these requested changes?

@alalonde Yes, finishing soon. Currently I am occupied with the hackantos stuff!

**Comment by dysbulic** [comment]
@arsenijesavic, do you plan to finish this or should I pick it up?

**Comment by arsenijesavic** [comment]
> @arsenijesavic, do you plan to finish this or should I pick it up?

@dysbulic yes, finishing this week

**Comment by arsenijesavic** [comment]
@dysbulic If you can review it now

**Pull request merged**